### PR TITLE
Add password confirmation field on registration page

### PR DESCRIPTION
**Pull request by dnguy24**

### CODE REVIEW

File: lms/static/js/student_account/views/RegisterView.js [hunk 1]
```diff
@@ -277,6 +278,9 @@
                     $('#register-confirm_email').bind('cut copy paste', function(e) {
                         e.preventDefault();
                     });
+                    $('#register-confirm_password').bind('cut copy paste', function(e) {
```
Comment (tuanh118) [inline lms/static/js/student_account/views/RegisterView.js:281]: Can we prevent "confirm password" field from pasting to enforce manual typing?

Comment (dnguy24) [reply lms/static/js/student_account/views/RegisterView.js:281]: @tuanh118 it is exactly what this line does

File: openedx/core/djangoapps/user_api/accounts/api.py [hunk 2]
```diff
@@ -579,6 +584,11 @@ def _validate_password(password, username=None, email=None):
         raise errors.AccountPasswordInvalid(' '.join(validation_err.messages))
 
 
+def _validate_confirm_password(confirm_password, password):
+    if not confirm_password or confirm_password != password:
```
Comment (hoanghiep106) [inline openedx/core/djangoapps/user_api/accounts/api.py:588]: This check `not confirm_password` should have a separate error message. Or if you required the field in another step, you can ignore this condition.

Comment (dnguy24) [reply openedx/core/djangoapps/user_api/accounts/api.py]: This part validates the confirm password field if the user moves their cursor outside of the box. Updated in latest commit to have a more informative message when the input is blank

File: openedx/core/djangoapps/user_api/accounts/api.py [hunk 3]
```diff
@@ -585,7 +585,9 @@ def _validate_password(password, username=None, email=None):
 
 
 def _validate_confirm_password(confirm_password, password):
-    if not confirm_password or confirm_password != password:
+    if not confirm_password:
+        raise errors.AccountPasswordInvalid('Please enter your Confirm password.')
```
Comment (hoanghiep106) [inline openedx/core/djangoapps/user_api/accounts/api.py:589]: Please enter your Confirm password. -> Please confirm your password. / Password confirmation is required

Comment (dnguy24) [reply openedx/core/djangoapps/user_api/accounts/api.py]: fixedd

**Pull request merged**